llvm/lib/Target/AMDGPU/SIInsertHardClauses.cpp

```diff
@@ -185,11 +191,18 @@ class SIInsertHardClauses {
   }
 
   bool run(MachineFunction &MF) {
-
     ST = &MF.getSubtarget<GCNSubtarget>();
     if (!ST->hasHardClauses())
       return false;
 
+    unsigned MaxClauseLength = MF.getFunction().getFnAttributeAsParsedInteger(
+        "amdgpu-hard-clause-length-limit", HardClauseLengthLimit);
```

Comment:
Suggest just writing the default here, instead of giving the option a `cl::init`.
```suggestion
        "amdgpu-hard-clause-length-limit", 255);
```